src/app/+orders/services/order-manager.service.ts

```diff
@@ -0,0 +1,125 @@
+import { Injectable } from '@angular/core';
+import { MatDialog, MatDialogRef } from '@angular/material';
+import { Store } from '@ngrx/store';
+import { AccountFacade } from 'facades/account.facade';
+import { CustomerFacade } from 'facades/customer.facade';
+import { OrderItemFacade } from 'facades/order-item.facade';
+import { OrderFacade } from 'facades/order.facade';
+import { ProductFacade } from 'facades/product.facade';
+import { OrderItem } from 'models/order-item.model';
+import { Order, OrderStatus } from 'models/order.model';
+import { combineLatest, Observable } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { omitByKeys } from 'shared/libs/util.lib';
+import {
+  IOrderFormDialogData,
+  OrderFormDialogComponent
+} from 'src/app/+orders/exports/order-form-dialog/order-form-dialog.component';
+import { IOrderFormItem, IOrderFormValue } from 'src/app/+orders/exports/order-form/order-form.component';
+import { OrderInfo } from 'src/app/+orders/models/order-info.model';
+import { AppState } from 'state/app.state';
+import { upsertFullOrder } from 'state/order.state';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OrderManagerService {
+  constructor(
+    private store: Store<AppState>,
+    private orderFacade: OrderFacade,
+    private customerFacade: CustomerFacade,
+    private orderItemFacade: OrderItemFacade,
+    private productFacade: ProductFacade,
+    private accountFacade: AccountFacade,
+    private dialogService: MatDialog
+  ) {}
+
+  /* Order Info */
+  orderInfoByStatus$(status$: Observable<OrderStatus[]>): Observable<OrderInfo[]> {
+    return status$.pipe(
```

Comment:
Another thing for future refactors of the demo app...we can probably do this work in selectors, rather than by piping off of other selectors. I think we could probably break down some of this work into multiple selectors, simplifying the code overall, but also leveraging the inherent momoization of selectors to improve performance.

Reply:
Yeah, this is definitely one of the ugly parts that I'd like to refactor for sure. I'll check to see if it's doable within a reasonable timeframe.